src/compiler/interpreter.jl

```diff
@@ -139,9 +152,22 @@ enzyme_inlining_policy(@nospecialize(src)) = Core.Compiler.default_inlining_poli
 Core.Compiler.inlining_policy(::EnzymeInterpeter) = enzyme_inlining_policy
 function Core.Compiler.resolve_todo(todo::InliningTodo, state::InliningState{S, T, <:typeof(enzyme_inlining_policy)}) where {S<:Union{Nothing, Core.Compiler.EdgeTracker}, T}
     mi = todo.mi
-    if is_primitive_func(mi.specTypes)
+    if is_primitive_func(mi.specTypes) 
+        return Core.Compiler.compileable_specialization(state.et, todo.spec.match)
+    end
+    if EnzymeRules.has_frule(mi.specTypes) || EnzymeRules.has_rrule(mi.specTypes)
         return Core.Compiler.compileable_specialization(state.et, todo.spec.match)
     end
+    ## TODO @vchuravy separate by mode like below
```

Comment:
@vchuravy how might I extract the interp (and thus mode) here?

Reply:
Maybe @aviatesk knows of the top of his head

Reply:
I think it's `state.interp`.